**This notebook is an exercise in the [Intro to Programming](https://www.kaggle.com/learn/intro-to-programming) course.  You can reference the tutorial at [this link](https://www.kaggle.com/alexisbcook/arithmetic-and-variables).**

---



**このノートは[プログラミング入門](https://www.kaggle.com/learn/intro-to-programming)コースの練習問題です。
[このリンク](https://www.kaggle.com/alexisbcook/arithmetic-and-variables)のチュートリアルを参照してください。**

---

This exercise will get you started with running your own code.  

# Set up the notebook

To begin, run the code in the next cell.
- Begin by clicking inside the code cell.  
- Click on the triangle (in the shape of a "Play button") that appears to the left of the code cell.
- If your code was run sucessfully, you will see `Setup Complete` as output below the cell.

Instead of clicking on the triangle, you can also run code by pressing Shift + Enter on your keyboard.  Try this now!  Nothing bad will happen if you run the code more than once.



---
この練習は、あなた自身のコードを実行することから始める。 

# ノートブックのセットアップ

始めるには、次のセルのコードを実行してください。
- コードのセル内をクリックして始めます。 
- コードセルの左側に表示される三角形（「再生ボタン」の形）をクリックしてください。
- コードがうまく実行されると、セルの下に`Setup Complete`と出力されます。

三角形をクリックする代わりに、キーボードのShift + Enterキーを押してコードを実行することもできます。 今すぐ試してみてください！ コードを複数回実行しても悪いことは起こりません。

In [21]:
# 練習の設定
from learntools.core import binder
binder.bind(globals())
from learntools.intro_to_programming.ex1 import *
print('Setup complete.')

Setup complete.


The code above sets up the notebook so that it can check your answers in this exercise.  You should never modify this code.  (Otherwise, the notebook won't be able to verify that you have successfully completed the exercise.)

After finishing all of the questions below, you'll see the exercise marked as complete on the [course page](http://www.kaggle.com/learn/intro-to-programming).  Once you complete all of the lessons, you'll get a course completion certificate!

# Question 1

Next, you will run some code from the tutorial, so you can see how it works for yourself.  Run the next code cell without changes.



---
上のコードは、この練習問題であなたの答えをチェックできるようにノートブックをセットアップするものです。 このコードは決して変更してはいけません。 (そうしないと、ノートブックはあなたが練習問題を正常に完了したことを確認できません)。

以下の問題をすべて解き終わると、[コースページ](http://www.kaggle.com/learn/intro-to-programming) に演習完了のマークが表示されます。 すべてのレッスンを完了すると、コース完了証明書が発行されます！

# 問題1

次に、チュートリアルのコードを実行します。 次のコードセルを変更せずに実行してください。

In [23]:
print("Hello, world!")

# 削除しないでください：この質問を完了としてマークしてください。
q1.check()

Hello, world!


<IPython.core.display.Javascript object>

<span style="color:#33cc33">If you see 'Hello, world!' above, You have successfully printed a message, and you're ready to move on to the next question.</span>

上記の'Hello, world!'が表示されていれば、メッセージのプリントは成功です。

You just ran code to print `Hello world!`, which you should see in the output above.

The second line of code (`q1.check()`) checks your answer.  You should never modify this checking code; if you remove it, you won't get credit for completing the problem.

# Question 2

Now, you will print another message of your choosing.  To do this, change `print("Your message here!")` to use a different message.  For instance, you might like to change it to something like:
- `print("Good morning!")`
- `print("I am learning how to code :D")`

Or, you might like to see what happens if you write something like `print("3+4")`.  Does it return 7, or does it just think of `"3+4"` as just another message?

Make sure that your message is enclosed in quotation marks (`"`), and the message itself does not use quotation marks. For instance, this will throw an error: `print("She said "great job" and gave me a high-five!")` because the message contains quotation marks.  If you decide to take the Python course after completing this course, you will learn more about how to avoid this error in [Lesson 6](https://www.kaggle.com/colinmorris/strings-and-dictionaries).

Feel free to try out multiple messages!



---
あなたは今、`Hello world!` を表示するコードを実行した。

2行目のコード（`q1.check()`）はあなたの答えをチェックします。 このチェックコードは決して変更してはいけません。もし削除してしまうと、問題を完了しても単位はもらえません。

# 質問2

ここで、あなたが選んだ別のメッセージを表示します。 これを行うには、`print("Your message here!")` を別のメッセージに変更してください。 例えば、次のようなメッセージに変更します：
- print("おはようございます!")
- print("私はコードを勉強しています :D")` のように。

あるいは、`print("3+4")`のように書いたらどうなるか見てみたいかもしれない。 print("3+4")`は7を返すのでしょうか？それとも`"3+4"`をただのメッセージとみなすのでしょうか？

メッセージが引用符(`"`)で囲まれていて、メッセージ自体が引用符を使っていないことを確認してください。例えば、`print("She said "great job" and gave me a high-five!")`とすると、メッセージに引用符が含まれているのでエラーになります。 このコースを修了した後、Pythonコースを受講することを決めた場合、このエラーを回避する方法について[Lesson 6](https://www.kaggle.com/colinmorris/strings-and-dictionaries)で詳しく学びます。

自由に複数のメッセージを試してみてください！


In [24]:
# TODO: メッセージを変更する
print("Your message here!")

# 削除しないでください： この質問を完了としてマークしてください。
q2.check()

Your message here!


<IPython.core.display.Javascript object>

<span style="color:#33cc33">Once you have printed your own message, you're ready to move on to the next question.</span>

あなたのメッセージをここに
自分のメッセージを印刷したら、次の質問に進みましょう。

# Question 3

As you learned in the tutorial, a comment in Python has a pound sign (`#`) in front of it, which tells Python to ignore the text after it.

Putting a pound sign in front of a line of code will make Python ignore that code.  For instance, this line would be ignored by Python, and nothing would appear in the output:
```python
#print(1+2)
```
Removing the pound sign will make it so that you can run the code again. When we remove the pound sign in front of a line of code, we call this **uncommenting**.

In this problem, you will uncomment two lines in the code cell below and view the output:
- Remove the `#` in front of `q3.hint()`.  To avoid errors, do NOT remove the `#` in front of `# Uncomment to view hint`.  
- Next, remove the `#` in front of `q3.solution()`.

As in the previous questions, do not change the final line of code that marks your work as completed.



---

# 質問 3

チュートリアルで学んだように、Pythonのコメントには、その前にポンド記号(`#`)があり、これはPythonにそれ以降のテキストを無視するように指示します。

コード行の前にポンド記号を置くと、Pythonはそのコードを無視します。 例えば、次の行はPythonに無視され、出力には何も表示されません：

```Python
#print(1+2)
```
ポンド記号を取り除くと、コードを再び実行できるようになります。コードの行の前にあるポンド記号を取り除くことを **コメント除去** と呼びます。

この問題では、下のコード・セル内の2行のコメントを解除して、出力を見てみましょう：
- `q3.hint()` の前にある `#` を削除する。 エラーを避けるため、`# Uncomment to view hint`. の前の `#` は削除しないでください。 
- 次に、`q3.solution()`の前の `#` を削除する。

前の問題と同様に、あなたの仕事が完了したことを示すコードの最後の行は変更しないでください。

In [25]:
# ヒントを得るためにコメントを外す
#q3.hint()

# 解答を表示するにはコメントを外す
#q3.solution()

# 削除禁止：答えをチェックする
q3.check()

<IPython.core.display.Javascript object>

<span style="color:#33cc33">Once you have printed the hint and the solution, you're ready to move on to the next question.</span>

ヒントと解答を印刷したら、次の問題に進む準備ができた。

In the next question, and in most of the exercises in this course, you will have the option to uncomment to view hints and solutions.  Once you feel comfortable with uncommenting, continue to the next question.




---
次の問題、そしてこのコースのほとんどの練習問題では、ヒントや解答を見るためにコメントを解除するオプションがあります。 コメント解除に慣れたら、次の問題に進んでください。

# Question 4

In the tutorial, you defined several variables to calculate the total number of seconds in a year.  Run the next code cell to do the calculation here.



---

# 第4問

チュートリアルでは、1年間の合計秒数を計算するためにいくつかの変数を定義しました。 ここで計算を行うために、次のコードセルを実行してください。

In [26]:
# 変数の作成
num_years = 4
days_per_year = 365 
hours_per_day = 24
mins_per_hour = 60
secs_per_min = 60

# 4年間の秒数を計算する
total_secs = secs_per_min * mins_per_hour * hours_per_day * days_per_year * num_years
print(total_secs)

126144000


Use the next code cell to:
- Define a variable `births_per_min` and set it to 250.  (There are on average 250 babies born each minute.)
- Define a variable `births_per_day` that contains the average number of babies born each day.  (To set the value of this variable, you should use `births_per_min` and some of the variables from the previous code cell.)

Remember you can always get a hint if you need it!



---
次のコードセルを使う：
- 変数 `births_per_min` を定義し、それを 250 に設定する。 (1分間に平均250人の赤ちゃんが生まれる)。
- 変数 `births_per_day` を定義し、毎日生まれる赤ちゃんの平均数を格納する。 (この変数の値を設定するために、`births_per_min`と前のコードのいくつかの変数を使う必要がある)。

必要であればいつでもヒントを得られることを忘れないでください！

In [27]:
# TODO: births_per_min変数の値を設定する
births_per_min = 250

# TODO: births_per_day変数の値を設定する
births_per_day = births_per_min * mins_per_hour * hours_per_day

# 答えをチェックする
q4.check()

<IPython.core.display.Javascript object>

<span style="color:#33cc33">Correct</span>

In [28]:
# ヒントを得るためにコメントを外す
#q4.hint()

# コメントを解除して解答を表示
#q4.solution()

# 🌶️ Question 5

(Questions marked with a 🌶️ will be a little bit more challenging than the others!  Remember you can always get a hint or view the solution.)

The [Titanic competition](https://www.kaggle.com/c/titanic) is Kaggle's most famous data science competition. In this competition, participants are challenged to build a machine learning model that can predict whether or not passengers survived the Titanic shipwreck, based on information like age, sex, family size, and ticket number.

Run the next code cell without changes to load and preview the titanic data.

Don't worry about the details of the code for now - the end result is just that the all of the titanic data has been loaded in a variable named `titanic_data`.  (In order to learn how to write this code yourself, you can take the [Python course](https://www.kaggle.com/learn/python) and then the [Pandas course](https://www.kaggle.com/learn/pandas).)



---
# 🌶️ 第5問

(🌶️、他の問題より少し難しくなっています！ いつでもヒントを得るか、解答を見ることができることを忘れないでください)。

[タイタニックコンペティション](https://www.kaggle.com/c/titanic)はKaggleで最も有名なデータサイエンスコンペティションです。このコンペティションでは、参加者は、年齢、性別、家族の人数、チケット番号などの情報に基づいて、タイタニック号の難破船から乗客が生還したかどうかを予測できる機械学習モデルを構築することに挑戦します。

次のコード・セルを変更せずに実行し、タイタニックのデータをロードしてプレビューしてください。

最終的な結果は、タイタニック号の全データが`titanic_data`という変数にロードされたということだけです。 (このコードの書き方を自分で学ぶには、[Pythonコース](https://www.kaggle.com/learn/python)を受講し、次に[Pandasコース](https://www.kaggle.com/learn/pandas)を受講すればよい)

In [31]:
# タイタニック大会からデータをロードする
import pandas as pd
titanic_data = pd.read_csv("../input/titanic/train.csv")

# データの最初の5行を表示する
titanic_data.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

The data has a different row for each passenger.  

The next code cell defines and prints the values of three variables:  
- `total` = total number of passengers who boarded the ship
- `survived` = number of passengers who survived the shipwreck
- `minors` = number of passengers under 18 years of age

Run the code cell without changes.  (Don't worry about the details of how these variables are calculated for now.  You can learn more about how to calculate these values in the [Pandas course](https://www.kaggle.com/learn/pandas).)



---
このデータには、乗客ごとに異なる行があります。 

次のコード・セルでは、3つの変数の値を定義して表示している：  
- `total` = 乗船した乗客の総数
- `survived` = 難破船から生還した乗客数
- `minors` = 18歳未満の乗客の数

コード・セルを変更せずに実行する。 (これらの変数がどのように計算されるかについては、今は気にしないでください。 これらの値の計算方法については[Pandasコース](https://www.kaggle.com/learn/pandas)で詳しく学ぶことができます)。

In [32]:
# 乗客総数
total = len(titanic_data)
print(total)

# 生存した乗客の数
survived = (titanic_data.Survived == 1).sum()
print(survived)

# 18歳未満の乗客数
minors = (titanic_data.Age < 18).sum()
print(minors)

891
342
113


So, 
- `total = 891` (there were 891 passengers on board the Titanic),
- `survived = 342` (342 passengers survived), and 
- `minors = 113` (113 passengers were under the age of 18).

In the code cell below, replace the underlines (`____`) with code to calculate the values for two more variables:
- `survived_fraction` should be set to the fraction of passengers who survived the Titanic disaster.
- `minors_fraction` should be the fraction of passengers who were minors (under the age of 18).

For each variable, your answer should be a number between 0 and 1.  

If you need a hint or want to view the solution, you can skip to the next code cell and uncomment the appropriate lines of code (`q5.hint()` and `q5.solution()`).



---
つまり 
- `total ＝ 891`（タイタニック号には891人の乗客が乗っていた）、
- `survived ＝ 342人`（342人の乗客が生き残った）、そして 
- `minors= 113`（113人の乗客が18歳未満であった）。

下のコードセルで、下線(`____`)を、さらに2つの変数の値を計算するコードに置き換える：
- `survived_fraction`には、タイタニック号の事故で生き残った乗客の割合を設定する。
- `minors_fraction`は、未成年（18歳未満）の乗客の割合とする。

各変数について、答えは0から1の間の数字でなければなりません。 

ヒントが必要な場合や解答を見たい場合は、次のコードセルに飛んで、適切なコード行（`q5.hint()`と `q5.solution()`）のコメントを解除してください。

In [33]:
# TODO：変数survived_fractionの値を埋める
survived_fraction = survived / total

# 変数の値を表示する
print(survived_fraction)

# TODO: minors_fraction変数の値を埋める
minors_fraction = minors / total

# 変数の値を表示する
print(minors_fraction)

# 答えをチェックする
q5.check()

0.3838383838383838
0.12682379349046016


<IPython.core.display.Javascript object>

<span style="color:#33cc33">Correct</span>

In [34]:
# ヒントを受け取るためにコメントを外す
q5.hint()

# コメントを解除して解答を見る
q5.solution()

<IPython.core.display.Javascript object>

<span style="color:#3366cc">Hint:</span> To get the fraction of people who survived the titanic, you need to divide the number of survivors by the total number of people.  Remember the variables that you can use to answer this question: `survived`, `total`, and `minors`.

<IPython.core.display.Javascript object>

<span style="color:#33cc99">Solution:</span> 
```python
# Fill in the value of the survived_fraction variable 
survived_fraction = survived/total

# Print the value of the survived_fraction variable
print(survived_fraction)

# Fill in the value of the minors_fraction variable
minors_fraction = minors/total 

# Print the value of the minors_fraction variable
print(minors_fraction)

```

ヒント：タイタニック号から生還した人の割合を求めるには、生還者数を総人数で割る必要がある。この問題に答えるために使える変数：生存者，合計，未成年者を覚えておこう．

解答

\#変数survived_fractionの値を記入する。  
survived_fraction = survived/total

\#変数survived_fractionの値を表示する  
print(survived_fraction)

\#変数minors_fractionの値を記入する  
minors_fraction = minors/total 

\#minors_fraction 変数の値を表示する  
print(minors_fraction)

# Bonus Exercise

As an **optional** next step, you're encouraged to make a submission to the Titanic competition.  
> To do this, follow the instructions in the **[notebook here](https://www.kaggle.com/alexisbcook/titanic-tutorial)**.  

The notebook is beginner-friendly and does not assume you have any experience with coding.  At the same time, you'll learn how to author your own notebooks on Kaggle, and it gives you a great idea of what you'll be able to do if you continue to learn data science!

# Keep going

Congratulations!  You have just finished the first exercise in the Intro to Programming course!  

You are now ready to learn how to **[organize your code with functions](https://www.kaggle.com/alexisbcook/functions)**.



---
# ボーナスエクササイズ

次のステップとして、タイタニックのコンペティションに応募してください。 
> そのためには、**[ノートブックはこちら](https://www.kaggle.com/alexisbcook/titanic-tutorial)**の指示に従ってください。 

このノートブックは初心者に優しく、コーディングの経験があることを前提としていません。 同時に、Kaggleで自分のノートブックを作成する方法を学ぶことができ、データサイエンスを学び続ければ何ができるようになるかを知ることができます！

# 続ける

おめでとうございます！ プログラミング入門コースの最初の演習が終わりました！ 

これで、**[関数を使ってコードを整理する](https://www.kaggle.com/alexisbcook/functions)**方法を学ぶ準備ができました。

---




*Have questions or comments? Visit the [course discussion forum](https://www.kaggle.com/learn/intro-to-programming/discussion) to chat with other learners.*



---

*質問やコメントがありますか？[コースディスカッションフォーラム](https://www.kaggle.com/learn/intro-to-programming/discussion)で他の学習者とチャットしてください。